sdk/search/search/src/searchIndexClient.ts

```diff
@@ -120,9 +120,13 @@ export class SearchIndexClient<T> {
         loggingOptions: {
           logger: logger.info,
           allowedHeaderNames: [
-            "x-ms-correlation-request-id",
-            "x-ms-request-id",
-            "client-request-id"
+            "elapsed-time",
+            "Location",
+            "OData-MaxVersion",
+            "OData-Version",
+            "Origin",
+            "Prefer",
+            "throttle-reason"
```

Comment:
Either this list needs to include `return-client-request-id` (until Search supports `x-ms-return-client-request-id` in a month or so), or it needs to go in core.

Reply:
updated